### PR TITLE
Null-safety migration

### LINKED ISSUE #37: Migration to null-safety
Starting from Flutter 2.0 and Dart SDK 1.12.0, dart language now supports sound null safety which requires explicitly defining null values.

### PR DESCRIPTION
**Pull request by TheMisir**
Fixes #37

### CODE REVIEW

**Comment by TheMisir** [comment]
Actually I think nnbd version should be maintained separately on different branch for some time. At least that's what we did when migrating [hive](https://github.com/hivedb/hive) to nnbd.

**Comment by curquiza** [comment]
Hello @TheMisir! Thanks for doing this work :)
Ping me when you have finished this PR 🙂

**Comment by TheMisir** [comment]
@curquiza it's ready. Just updated `pubspec.yaml` version to v0.2.0 and tests are passing (if dart >2.12).

**Review by curquiza — CHANGES_REQUESTED** [review]
Thanks a lot @TheMisir 😄

- Can you update the [bors.toml](https://github.com/meilisearch/meilisearch-dart/blob/main/bors.toml) file please? Some tests have to be removed 😇
- Does the README need changes? I think about the `Getting Started` section

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
       - trying
       - staging
       - main
+      - nnbd
```
Comment (curquiza) [inline .github/workflows/tests.yml:11]:
```suggestion
```

**Review by curquiza — CHANGES_REQUESTED** [review]

**Comment by TheMisir** [comment]
> * Can you update the [bors.toml](https://github.com/meilisearch/meilisearch-dart/blob/main/bors.toml) file please? Some tests have to be removed 😇

I've updated `bors.toml` because testing requires dart ^2.12.

**Comment by curquiza** [comment]
> I've updated bors.toml because testing null safety requires dart ^2.12.

My bad, I miss checked! :)

**Review by curquiza — APPROVED** [review]
Awesome!! 🔥 Thanks @TheMisir!

**Comment by curquiza** [comment]
bors merge

**Comment by TheMisir** [comment]
@curquiza out of context, but what does **bors** exactly?

**Comment by curquiza** [comment]
TLDR: Bors is a tool that applies a rebase before running the tests and finally merging if they succeed.

To be more accurate, Bors does not concretely rebase the branch of the PR: it merges the commits of the PR into another branch (staging) that is already up-to-date with main. If the tests do not pass on the staging branch, Bors returns a failure and stops the merge of the PR.

We use it because:
- we don't have to ask the contributors to rebase anymore which is a really nice point 😁
- it can handle multiple PRs at the same time: no need to wait for the PR of your colleague to be merged to merge yours.

I wrote a guide about it here: https://github.com/meilisearch/integration-guides/blob/main/guides/bors.md
Also, I wrote an article about our usage of bors, but it's not released yet 😇 

Also, bors is a tool widely used (and created) by the rust community

**Comment by TheMisir** [comment]
Thanks 🙏
